### DISCUSSION TITLE
ROI "operation"

### DISCUSSION THREAD

**Discussion by Ashwolaa** [PyMoDAQ/PyMoDAQ#192] [Ideas]
The ROI operation are great to follow the signal and optimize a particular region. 
However, the current viewer does not allow a few features that would make it much better. I believe these features already exists in the DAQ_Viewer0D.

- There is no choice concerning the maximum number of iterations to be saved.
- There is no possibility to zoom which makes the display useless when there are small variations when compared to large ones.
- There is no reset button
- There is no y axis? I am not sure to understant that, is it because you can have different operations? Then each operation could be attached to a viewer instead? or use a left/righ axis when there are only two
- Also, would it be possible to make the ROI apply on multiple channels? It seems that I have to pick the curve I want to do the operation on while sometimes I would like to apply it on all the curves at the same time.

**Comment by seb5g** [comment]
Quoting your issue:
* There is no choice concerning the maximum number of iterations to be saved.
* 
what do you mean?

* There is no possibility to zoom which makes the display useless when there are small variations when compared to large ones.
* There is no reset button
* There is no y axis? I am not sure to understant that, is it because you can have different operations? Then each operation could be attached to a viewer instead? or use a left/righ axis when there are only two

I guess from that you are refering to the 1D data viewer? I yes then you are right... I should at lest have the same as for the 2D viewer lineouts. I'll have a look.

* Also, would it be possible to make the ROI apply on multiple channels? It seems that I have to pick the curve I want to do the operation on while sometimes I would like to apply it on all the curves at the same time.

Could you not just specify multiple ROI for different curves?

Generally speaking the plot in the lineouts are quite basics ones there are not themselves 0D (from roi of a 1D viewer) or 1D viewers (from roi of a 2D viewer). I'm not sure it would be a good idea to use a full feature 0D viewer for this but I can try. For this could you specify a more detailed issue like: 1D Data Viewer: implementing a 0D data viewer for lineouts (or somehting like that) because here I had to guess about what you were talking about (ROI and lineouts for 1D data viewers)
